src/core/config/from_document.rs

```diff
@@ -422,6 +428,22 @@ fn to_add_fields_from_directives(
     )
 }
 
+fn to_unknown_directives(
+    directives: &[Positioned<ConstDirective>],
+) -> Valid<Vec<Directive>, String> {
+    Valid::from_iter(directives.iter(), |directive| {
+        if KNOWN_DIRECTIVES
```

Comment:
This potentially is going to introduce a bug when the developer forgets to update known_directives. We should change the logic to detect directives here. Keep a mutable list of directives and keep removing the consumed directives and add the remaining as unknown (or something around those lines)

Reply:
I agree this approach is error-prone. While doing this I explored the other possibilities but found it'll will either require some significant refactoring around directives or may introduce other bugs since it'll require more code to work. 

So I went by simple path since I saw already similar approach https://github.com/tailcallhq/tailcall/blob/aab7e07349cfb6421c231260e40bd4a97e7ea80b/src/core/config/config.rs#L681-L733

Reply:
What would be your suggestion here? I think directives have also other flaws that could be reworked together (e.g. working only with non repeatable directives, no way to provide custom implementation of trait DirectiveCodec etc.)

Reply:
Alternative approach: How about we create a white list of additional directives that we can copy. For eg: If `enableFederation` is set to true, we allow the federation specific directives into the configuration.